### PR TITLE
Change branch master to main

### PR DESCRIPTION
**Pull request by adangel**
## Describe the PR

This applies the changes so that we can switch to main. I'd suggest to make this change already for the next version (7.6.0).

- Update documentation
- Update release scripts
- Update rule doc generation

GitHub Action is only triggered from "main" branch now.

## Note in release notes:

----

#### New Git default branch - "main"

We are joining the Git community and updating "master" to "main". Using the term "master" for the main
development branch can be offensive to some people. Existing versions of Git have been always capable of
working with any branch name and since 2.28.0 (July 2020) the default initial branch is configurable
(`init.defaultBranch`). Since October 2020, the default branch for new repositories on GitHub
is "main". Finally, PMD will also use this new name for the main branch in all our own repositories.

Why "main"? PMD uses a very simple branching model - pull requests with feature branches and one main development
branch, from which releases are created. That's why "main" is currently the best fitting name.

More information:
- <https://sfconservancy.org/news/2020/jun/23/gitbranchname/>
- <https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/>

What changes?
- We change the default branch on GitHub, so that pull requests are automatically created against `main` from
  now on.
- If you have already a local clone of PMD's repository, you'll need to rename the old master branch locally:
  ```
  git branch --move master main
  git fetch origin
  git branch --set-upstream-to=origin/main main
  git remote set-head origin --auto
  ```
  
  More info:
  <https://git-scm.com/book/en/v2/Git-Branching-Branch-Management#_changing_master> and
  <https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch#updating-a-local-clone-after-a-branch-name-changes>
- If you created a fork on GitHub, you'll need to change the default branch in your fork to `main` as
  well (Settings > Default Branch).
- Some time after this release, we'll delete the old master branch on GitHub. Then only `main` can be used.
- This change is expanded to the other PMD repositories as well, e.g. pmd-designer and pmd-regression-tester.

----

## Related PRs

- pmd/pmd-eclipse-plugin#220
- pmd/build-tools#34
- pmd/pmd-designer#93
- pmd/pmd-regression-tester#128
- :heavy_check_mark: [pmd-examples](https://github.com/pmd/pmd-examples) (just deleted old master and other old branches)
- :heavy_check_mark: [pmd.github.io](https://github.com/pmd/pmd.github.io)
- :heavy_check_mark: [pmd.sourceforge.net](https://github.com/pmd/pmd.sourceforge.net)
- pmd/pmd-bluej#9
- :heavy_check_mark: https://github.com/pmd/pmd-scm/commit/7a9eb5c378d92f4116d54151debb8e9fb72e9418
- :heavy_check_mark: [pmd-emacs](https://github.com/pmd/pmd-emacs)
- :heavy_check_mark: [pmd-misc](https://github.com/pmd/pmd-misc)
- :heavy_check_mark: [pmd-jdeveloper](https://github.com/pmd/pmd-jdeveloper)

## Ready?

The PR is in draft, because I want to make similar changes for our other repositories.

- [ ] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://github.com/pmd/pmd/actions/runs/10845897438?pr=5211">Download full report as build artifact</a></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to main:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://github.com/pmd/pmd/actions/runs/10845897438?pr=5211">Download full report as build artifact</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: docs/pages/release_notes.md [hunk 1]
```diff
@@ -14,6 +14,30 @@ This is a {{ site.pmd.release_type }} release.
...
+More information:
+- <https://sfconservancy.org/news/2020/jun/23/gitbranchname/>
+- <https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/>
+
+What changes?
+- We change the default branch on GitHub, so that pull requests are automatically created against `main` from
+  now on.
+- If you have already a local clone of PMD's repository, you'll need to rename the old master branch locally:
+  `git branch --move master main && git branch --set-upstream-to=origin/main`. More info:
+  <https://git-scm.com/book/en/v2/Git-Branching-Branch-Management#_changing_master>
```
Comment (adangel) [inline docs/pages/release_notes.md:37]:
```suggestion
- If you created a fork on GitHub, you'll need to change the default branch in your fork to `main` as well (Settings > Default Branch)
- If you have already a local clone of PMD's repository, you'll need to rename the old master branch locally:
  `git branch --move master main && git branch --set-upstream-to=origin/main`. More info:
  <https://git-scm.com/book/en/v2/Git-Branching-Branch-Management#_changing_master>
```

**Comment by adangel** [comment]
This is ready now.

I'm going to proceed with changing master -> main first for the other repos (build-tools, pmd-eclipse-plugin, pmd-designer, pmd-regression-tester).
As last, I'll change it for the main pmd repo.

Note: You'll have to adjust your local clone as well, see release notes.

**Comment by jsotuyod** [comment]
@adangel when should we remove the `master` branch altogether from the repo?

I fear we will keep merging PRs against it by mistake (specially already existing ones).

**Comment by adangel** [comment]
I've updated all the old PRs to point to `main` already. New PRs should automatically preselect `main`.

I planned to push to both `main` and `master` to keep them in sync for a while (just for convenience if someone still is on master) - but if that makes more trouble, then we can also just remove it right now...

**Pull request merged**